src/vsr/replica.zig

```diff
@@ -5357,13 +5451,14 @@ pub fn ReplicaType(
         }
 
         fn set_op_and_commit_max(self: *Self, op: u64, commit_max: u64, method: []const u8) void {
-            assert(self.status == .view_change);
+            assert(self.status == .view_change or self.status == .normal or
+                self.status == .recovering_head);
 
             // Uncommitted ops may not survive a view change so we must assert `op` against
             // `commit_max` and not `self.op`. However, committed ops (`commit_max`) must survive:
-            assert(op >= self.commit_max);
-            assert(op >= commit_max);
             assert(op <= self.op_checkpoint_trigger());
+            maybe(op >= self.commit_max);
+            maybe(op >= commit_max);
```

Comment:
Hm, the comment probably needs to be updated to match new assertions? Or maybe rather assertions needs to be updated to something like `if (op < self.op) assert(op >= max(commit_max, self.commit_max))`. That is, if we truncate, we never truncate commited stuff.